src/datasets/utils/filelock.py

```diff
@@ -96,6 +104,24 @@ def __str__(self):
         return temp
 
 
+# Helpers
+# ------------------------------------------------
+
+
+MAX_LOCK_FILENAME_LENGTH = 255
```

Comment:
Is this a parameter that the user should be allowed to customize? If yes, I would suggest moving it to `config`; if not (I think this is the case), then I would suggest to remove the variable and pass the value directly in the function signature of `hash_filename_if_too_long`.